db/permission/register.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package permission
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/acp"
+	"github.com/sourcenetwork/defradb/client"
+)
+
+// RegisterDocOnCollectionWithACP handles the registration of the document with acp module.
+// The registering is done at document creation on the collection.
+//
+// According to our access logic we have these components to worry about:
+// (1) the request is permissioned (has an identity signature),
+// (2) the collection is permissioned (has a policy),
+// (3) acp module exists (acp is enabled).
+//
+// The document is only registered if all (1) (2) and (3) are true.
+//
+// Otherwise, nothing is registered on the acp module.
+func RegisterDocOnCollectionWithACP(
+	ctx context.Context,
+	identity immutable.Option[string],
+	acpModule immutable.Option[acp.ACPModule],
```

Comment:
suggestion: I see the `permission` package as a util package to encapsulate some permission/acp functionality. As such, I don't think the params here should be `Options`. This is somewhat of a common theme to some of my other comments regarding `Option` in this PR. Since its a util package (feel free to assert otherwise) it should be the callers responsibility to determine if the util func needs to be called (ie: only when `acp` exists).

Reply:
I understand what you are saying, and thought about it from both these sides while I was implementing it. Since we handle the case of not having acp enabled (no acp module) somewhat differently, it plays a part in what it means to check or register for example. I like all the decision making logic in one place as much as possible, so callers have less work and logic states to check and can rely the decision making to be done in one place.

Reply:
Keeping as is, unless harder preference or consensus reaches to do this change.

Reply:
callers *should* be responsible though, if they are using a util package, that they have done the work to use properly use said util package. Otherwise the scope of util packages would very quickly explode. 

The whole point of them is to some direct utility function and nothing more. These funcs shouldnt be concerned if acp is or isnt enabled, they should do the action they describe using the inputs theyre given. There isnt a concept of an "optional" ACP from the POV of a util function imo.

Feels like its just doing it because the initial caller this was designed for had it, rather than the util function actually needing to handle different cases based on the `Option` type.

Reply:
> There isnt a concept of an "optional" ACP from the POV of a util function imo.

I wanted there to be a util function that can manage those concepts, otherwise I don't really even need this function and can completely remove it.

Since you have a stronger preference, I will remove `Option` type for acp, but keeping for identity.

- [ ] Remove optional acp on Registering util func